From 548c1a9613ac7f669601c7cc816690f1a44f99c0 Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Tue, 4 Feb 2020 02:45:57 +0100
Subject: [PATCH 1/2] sepolicy: Allow Snap to read camera properties

avc: denied { read } for name="u:object_r:camera_prop:s0" dev="tmpfs" ino=6304 scontext=u:r:snap_app:s0:c512,c768 tcontext=u:object_r:camera_prop:s0 tclass=file permissive=0 app=org.lineageos.snap

Change-Id: I89c95548904cfb9a5a50a8833f8e2a196eb59dcf
---
 common/private/snap_app.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/common/private/snap_app.te b/common/private/snap_app.te
index 40ecac8..969b3c4 100644
--- a/common/private/snap_app.te
+++ b/common/private/snap_app.te
@@ -35,3 +35,6 @@ allow snap_app self:process ptrace;
 
 # Read and write system app data files passed over Binder.
 allow snap_app system_app_data_file:file { read write getattr };
+
+# Allow to read camera properties
+get_prop(snap_app, camera_prop)
-- 
2.7.4

